provider crash if harbor url is missing

**Describe the bug**
The harbor provider crash if the harbor url configuration is missing
 

**To Reproduce**
unset environment variable HARBOR_URL or comment provider.harbor.url from provider config.

**Expected behavior**
The provider should error out with a meaning full error message

**Screenshots**
```
Stack trace from the terraform-provider-harbor_v3.7.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x89c905]

goroutine 11 [running]:
github.com/BESTSELLER/terraform-provider-harbor/client.(*Client).SendRequest(0xc0000b0a80, {0xc04b05, 0x3}, {0xc0005c9420?, 0xc0004fe5c0?}, {0x0, 0x0}, 0xc8)
	github.com/BESTSELLER/terraform-provider-harbor/client/client.go:64 +0x545
github.com/BESTSELLER/terraform-provider-harbor/provider.dataRegistryRead(0x7f922d6ad668?, {0xb7fba0?, 0xc0000b0a80})
	github.com/BESTSELLER/terraform-provider-harbor/provider/data_registry.go:53 +0xdd
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xd254c0?, {0xd254c0?, 0xc00070eb10?}, 0xd?, {0xb7fba0?, 0xc0000b0a80?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0001edce0, {0xd254c0, 0xc00070eb10}, 0xc000142b80, {0xb7fba0, 0xc0000b0a80})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc0003a8408, {0xd25418?, 0xc0004fe500?}, 0xc000718380)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:1179 +0x38f
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0002ccaa0, {0xd254c0?, 0xc00070e390?}, 0xc000710140)
	github.com/hashicorp/terraform-plugin-go@v0.9.1/tfprotov5/tf5server/server.go:657 +0x409
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0xbdade0?, 0xc0002ccaa0}, {0xd254c0, 0xc00070e390}, 0xc000182360, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.9.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002fc8c0, {0xd27f50, 0xc0000036c0}, 0xc0001ac480, 0xc0003be3c0, 0x1192590, 0x0)
	google.golang.org/grpc@v1.46.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc0002fc8c0, {0xd27f50, 0xc0000036c0}, 0xc0001ac480, 0x0)
	google.golang.org/grpc@v1.46.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.46.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.46.0/server.go:920 +0x28a

Error: The terraform-provider-harbor_v3.7.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/provider/provider.go
+++ b/provider/provider.go
@@ -1,6 +1,8 @@
 package provider
 
 import (
+	"fmt"
+	"os"
 	"strings"
 
 	"github.com/goharbor/terraform-provider-harbor/client"
@@ -74,7 +76,15 @@ func Provider() *schema.Provider {
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	var apiPath string
 
-	url := d.Get("url").(string)
+	//url := d.Get("url").(string)
+	url := os.Getenv("HARBOR_URL")
+	if d.Get("url").(string) != "" {
+		url = d.Get("url").(string)
+	}
+	if url == "" {
+		return nil, fmt.Errorf("url is required and must be provided in the provider config or the HARBOR_URL environment variable")
+	}
+
 	username := d.Get("username").(string)
 	password := d.Get("password").(string)
 	insecure := d.Get("insecure").(bool)

